src/api/MainSolver.cc

```diff
@@ -223,6 +223,13 @@ std::unique_ptr<Model> MainSolver::getModel() {
     return modelBuilder.build();
 }
 
+const ResolutionProof& MainSolver::getResolutionProof() const {
+    //? check smt_solver->logsResolutionProof() ?
```

Comment:
It would be good to resolve this comment.
There should be a way to check if we really have a proof.
Currently, `CoreSMTSolver::getResolutionProof` only uses `assert`, which is too weak.
I think checking `smt_solver->logsResolutionProof` should be sufficient. It can be conjoined to the `status != s_False` check.

Reply:
The reason why I did not implement it this way right away is that `getModel` also does not do such checks. To make it I consistent, I think an update where similar checks are added should follow. Do you agree?
And maybe it would be even better if both of these changes (i.e. of `getModel` and also `getResProof`) would be done in a single PR, that is, not here. And maybe also in the case of `getInterpolationContext`.

Reply:
`getModel` and `getInterpolationContext` are different, because they return unique pointers. They create these objects on demand. The situation with proof is different because we are returning a reference. So we need to make sure that we have the object.

Reply:
Ah, but I see that we should check that OpenSMT has been configured to produce models and proofs, respectively.